From 43219f1b6bd3cb538ac01a631226a471ec053637 Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Wed, 3 Feb 2021 18:45:19 +0100
Subject: [PATCH 1/1] UbuntuDrivers/detect.py: change metapackage generation

---
 UbuntuDrivers/detect.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/UbuntuDrivers/detect.py b/UbuntuDrivers/detect.py
index b6e4dee..f6678a3 100644
--- a/UbuntuDrivers/detect.py
+++ b/UbuntuDrivers/detect.py
@@ -1128,8 +1128,10 @@ def get_linux_modules_metapackage(apt_cache, candidate):
         # dependencies
         reverse_deps = find_reverse_dependencies(apt_cache, linux_modules_match, 'linux-modules-nvidia-')
         pick = ''
+        modules_candidate = 'linux-modules-nvidia-%s-%s' % (candidate_flavour,
+                                                            get_linux_image(apt_cache).replace('linux-image-', ''))
         for dep in reverse_deps:
-            if dep > pick:
+            if dep == modules_candidate:
                 pick = dep
         if pick:
             metapackage = pick
-- 
2.25.1

